src/wayland/shm/mod.rs

```diff
@@ -160,8 +160,11 @@ impl ShmState {
     /// Updates the list of formats advertised by the global.
     ///
     /// This will only affect new binds to the wl_shm global.
-    pub fn add_formats(&mut self, formats: impl IntoIterator<Item = wl_shm::Format>) {
-        self.formats.extend(formats.into_iter());
+    ///
+    /// Removing formats will cause old clients trying to create
+    /// a buffer of a now unsupported format to be killed.
+    pub fn update_formats(&mut self, formats: impl IntoIterator<Item = wl_shm::Format>) {
+        self.formats = formats.into_iter().collect::<HashSet<_>>();
```

Comment:
This could potentially remove the mandatory formats, which feels a bit inconsitent as `ShmState::new` always adds them also mentions in the docs that the global will always advertize them.